src/proof/nova.rs

```diff
@@ -237,22 +261,41 @@ impl<'a, F: CurveCycleEquipped, C: Coprocessor<F>> RecursiveSNARKTrait<F, C1LEM<
         steps: Vec<C1LEM<'a, F, C>>,
         store: &Store<F>,
     ) -> Result<Self, ProofError> {
-        assert!(!steps.is_empty());
-        assert_eq!(steps[0].arity(), z0.len());
         let debug = false;
-        let z0_primary = z0;
-        let z0_secondary = Self::z0_secondary();
+        assert_eq!(steps[0].arity(), z0.len());
 
-        let circuit_secondary = TrivialCircuit::default();
+        let secondary_circuit = TrivialCircuit::default();
 
         let num_steps = steps.len();
-        tracing::debug!("steps.len: {num_steps}");
+        info!("proving {num_steps} steps");
 
-        // produce a recursive SNARK
-        let mut recursive_snark: Option<RecursiveSNARK<E1<F>>> = None;
+        let mut recursive_snark_option: Option<RecursiveSNARK<E1<F>>> = None;
 
-        // the shadowing here is voluntary
-        let recursive_snark = if lurk_config(None, None)
+        // a closure would capture (and move) `recursive_snark_option`
```

Comment:
Could this closure not take an `|recursive_snark: &mut Option<RecursiveSNARK>|` argument instead of mentioning the `recursive_snark` and mutating it implicitly?

I'd rather avoid a macro here, for reasons of not really improving in complexity over the original.

Reply:
If I do that, it seems like I'm required to `clone` it before calling `unwrap_or_else`, which is one of the things I'm trying to avoid with this PR

Reply:
This closure will work nicely:
```rust
        let prove_step = |rs: &mut Option<RecursiveSNARK<E1<F>>>, i: usize, step: &C1LEM<'a, F, C>| {
            let mut recursive_snark = rs.take().unwrap_or_else(|| {
                RecursiveSNARK::new(
                    &pp.pp,
                    step,
                    &secondary_circuit,
                    z0,
                    &Self::z0_secondary(),
                )
                .expect("failed to construct initial recursive SNARK")
            });
            info!("prove_step {}", i);
            recursive_snark
                .prove_step(&pp.pp, step, &secondary_circuit)
                .unwrap();
            *rs = Some(recursive_snark);
        };
```